pkg/prediction/generic.go

```diff
@@ -57,27 +58,43 @@ func (p *GenericPrediction) WithProviders(providers map[string]providers.Interfa
 	}
 }
 
-func (p *GenericPrediction) WithQuery(query string) error {
-	if query == "" {
-		return fmt.Errorf("empty query")
+func (p *GenericPrediction) WithQuery(queryExpr string) error {
+	if queryExpr == "" {
+		return fmt.Errorf("empty query expression")
 	}
 
 	p.mu.Lock()
 	defer p.mu.Unlock()
 
-	if _, exists := p.querySet[query]; !exists {
-		p.querySet[query] = struct{}{}
-		p.withQueryBroadcaster.Write(query)
+	if _, exists := p.querySet[queryExpr]; !exists {
+		p.querySet[queryExpr] = struct{}{}
+		p.withCh <- queryExpr
 	}
 
 	return nil
 }
 
-func AggregateSignalKey(id string, labels []common.Label) string {
+func (p *GenericPrediction) DeleteQuery(queryExpr string) error {
+	if queryExpr == "" {
+		return fmt.Errorf("empty query expression")
+	}
+
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	if _, exists := p.querySet[queryExpr]; exists {
```

Comment:
Consider this situation:
EHPA A is running for a pod named "test1", and meanwhile Recommendation B is recommend hpa for the same pod "test1", their queryExpr for cpu usage is same. if delete the query here, then should effect the EHPA ?

Reply:
Add caller to WithQuery/DeleteQuery